encrypted-media/scripts/reset-src-after-setmediakeys.js

```diff
@@ -0,0 +1,61 @@
+function runTest(config)
+{
+    async_test(function (test) {
+        var mediaKeys;
+        var mediaSource;
+        var encryptedEventIndex = 0;
+        var video = config.video;
+        var keysystem = config.keysystem;
+        var configuration = {
+            initDataTypes: [config.initDataType],
+            audioCapabilities: [{
+                contentType: config.audioType
+            }],
+            videoCapabilities: [{
+                contentType: config.videoType
+            }],
+            sessionTypes: ['temporary']
+        };
+
+        assert_not_equals(video, null);
+
+        var onEncrypted = function (event) {
+            ++encryptedEventIndex;
+            assert_equals(video.mediaKeys, mediaKeys);
+
+            // This event is fired once for the audio stream and once
+            // for the video stream each time .src is set.
+            if (encryptedEventIndex === 2) {
+                // Finished first video; Create new media source and wait for two more encrypted events
+                return testmediasource(config).then(function (source) {
+                    video.src = URL.createObjectURL(source);
+                }).catch(function (error) {
+                    forceTestFailureFromPromise(test, error)
+                });
+            } else if (encryptedEventIndex === 4) {
+                // Finished second video.
+                test.done();
+            }
+        };
+
+        // Create a MediaKeys object and assign it to video.
+        return navigator.requestMediaKeySystemAccess(keysystem, [configuration]).then(function (access) {
```

Comment:
I will prefer to wrap the function in "then" with step_func. promise_test is difficult here since the event handler is called 4 times.